### PR TITLE
Support converting string literals to templates outside of concatenation

### PR DESCRIPTION
**Pull request by jakebailey**
This PR updates our "convert to template string" code action to support plain string literals. For example:

templatestring

This action is only available when the code fix is "invoked" meaning it won't get a lightbulb, unlike concat expressions.

There are still some edge cases that don't work thanks to the way this code action works. For example, in this code:

```ts
if (nodeIsStringLiteral && context.triggerReason !== "invoked") {
    return emptyArray;
}
```

Attempting to query on `"invoked"` will not give a quick fix because the binary expression that is found is the entire `nodeIsStringLiteral && context.triggerReason !== "invoked"`, which then doesn't look "valid" and so fails. Fixing this will take some restructuring and this PR does the simple thing which I think handles the bulk of the cases.

---

For a future PR, here's a test case that exhibits the edge case:

```ts
/// <reference path='fourslash.ts' />

//// if (false && "" !== '/*x*/f/*y*/oobar rocks') {}

goTo.select("x", "y");
verify.not.refactorAvailable(ts.Diagnostics.Convert_to_template_string.message);
verify.refactorAvailableForTriggerReason("invoked", ts.Diagnostics.Convert_to_template_string.message);
verify.not.refactorAvailableForTriggerReason("implicit", ts.Diagnostics.Convert_to_template_string.message);

edit.applyRefactor({
    refactorName: "Convert to template string",
    actionName: "Convert to template string",
    actionDescription: ts.Diagnostics.Convert_to_template_string.message,
    triggerReason: "invoked",
    newContent:
        `if (false && "" !== \`foobar rocks\`) {}`,
});
```

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot pack this

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -7512,7 +7512,7 @@
         "category": "Message",
         "code": 95153
     },
-    "Can only convert string concatenation": {
+    "Can only convert string concatenation or string literals": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json:7515]:
```suggestion
    "Can only convert string concatenations and string literals": {
```

**Comment by sandersn** [comment]
@jakebailey didn't we talk about this at a design meeting? If so, did we decide to take it?

**Comment by jakebailey** [comment]
We talked about at an editor sync but I don't think there were objections, but honestly I don't recall anyone having any comments at all... 😄

**Comment by DanielRosenwasser** [comment]
Have we confirmed (with tests) that we don't provide the refactoring in places where it would be invalid?

e.g.

```ts
import "foo/**/";

import * as x from "x" asserts { "lol/**/": "wut" };
import * as x from "x" with { "lol/**/": "wut" };

let x = {
  "foo/**/": 123,
}
```

**Comment by jakebailey** [comment]
No, definitely not! I have yet to refine this PR for the previous changes and I'll go test those out. Are there any more than just those?

**Comment by jakebailey** [comment]
I've updated the PR to disallow this refactor in non-expression contexts.

@typescript-bot pack this

**Review by weswigham — APPROVED** [review]

**Pull request merged**